specification/communication/resource-manager/Microsoft.Communication/preview/2023-06-01-preview/SuppressionLists.json

```diff
@@ -0,0 +1,642 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SuppressionListName"
+          },
+          {
+            "$ref": "#/parameters/SuppressionListAddressId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The resource was deleted successfully."
+          },
+          "204": {
+            "description": "The resource did not exist to be deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete a SuppressionListAddress resource.": {
+            "$ref": "./examples/suppressionLists/deleteAddress.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SuppressionListResource": {
+      "description": "A class representing a SuppressionList resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SuppressionListProperties",
+          "description": "The properties of a SuppressionList resource.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SuppressionListAddressResourceCollection": {
+      "description": "Collection of addresses in a suppression list. Response will include a nextLink if response contains more pages.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of suppressed email addresses.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SuppressionListAddressResource"
+          }
+        },
+        "nextLink": {
+          "description": "The URL the client should use to fetch the next page (per server side paging).",
+          "type": "string"
+        }
+      }
+    },
+    "SuppressionListProperties": {
+      "description": "A class that describes the properties of a SuppressionList resource.",
+      "type": "object",
+      "properties": {
+        "listName": {
+          "description": "The the name of the suppression list. This value must match one of the valid sender usernames of the sending domain.",
+          "type": "string"
+        },
+        "dataLocation": {
+          "description": "The location where the SuppressionList resource data is stored at rest.",
```

Comment:
>": "The location where the SuppressionList resource data is stored at rest.", [](http://example.com/codeflow?start=22&length=77)

A proxy resource is not supposed to have a location. Customers expect the resource to have the same location as the parent tracked resource and also arm will route this also accordingly. How do you enable this feature ?

Reply:
That dataLocation is inherited from the parent DomainsResource, we look that up at resource creation time.  

Should I indicate in the comments that this property reflects the parent domain's dataLocation, or should I remove this and keep it stored in our internal metadata (i.e. do not return to the customer)?

Reply:
I would suggest not returning this to the customer at all. It is implicit in the way that Proxy resources are defined that they inherit the location of their parent. 

If for some reason you really need it then make it readonly so that no one tries to set it. It is in general a good idea to indicate all the readonly fields in your swagger.

Reply:
Removed dataLocation and made the dateTime properties readonly